recipes/arm-gnu-toolchain/all/test_package/conanfile.py

```diff
@@ -0,0 +1,24 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, cmake_layout
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualBuildEnv"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def build(self):
+        if str(self.settings.compiler) == "gcc":
```

Comment:
This if smells iffy.
We need more insight into why this was put here, or at least logs of it working on Windows this is not tested by the CI (No mingw configurations are currently built, although it's a planned feature)

Also (and this is just me maybe missing something else) why is this check here and not on the validate() method of the recipe? Any usecase?